packages/sources/intrinio/src/config.ts

```diff
@@ -1,36 +1,61 @@
-import { util } from '@chainlink/ea-bootstrap'
+import { Requester, Validator } from '@chainlink/ea-bootstrap'
+import { MakeWSHandler, Config, AdapterRequest, AdapterResponse } from '@chainlink/types'
+import { customParams } from './adapter'
+import IntrinioRealtime from 'intrinio-realtime'
 
 /**
  * @swagger
  * securityDefinitions:
  *  environment-variables:
- *    API_PROVIDER:
- *      required: false
  *    API_KEY:
  *      required: true
- *    SYMBOLS:
- *      required: true
  */
 
 export const NAME = 'INTRINIO'
 
-export const PROVIDER_OPTIONS = ['iex', 'quodd', 'fxcm']
+export const DEFAULT_API_ENDPOINT = 'https://api-v2.intrinio.com/'
+// const DEFAULT_WS_API_ENDPOINT = ''
 
-export type Config = {
-  key: string
-  symbols: string
-  provider: string
+export const makeConfig = (prefix?: string): Config => {
+  const config = Requester.getDefaultConfig(prefix, true)
+  config.api.baseURL = config.api.baseURL || DEFAULT_API_ENDPOINT
+  return config
 }
 
-export const makeConfig = (prefix?: string): Config => {
-  let provider = util.getEnv('API_PROVIDER', prefix) || ''
-  if (!PROVIDER_OPTIONS.includes(provider)) {
-    provider = PROVIDER_OPTIONS[0]
+export const makeWSHandler = (config?: Config): MakeWSHandler => {
+  // https://github.com/intrinio/intrinio-realtime-node-sdk
+
+  const getBase = (input: AdapterRequest): string => {
+    const validator = new Validator(input, customParams)
+    if (validator.error) {
+      return ''
+    }
+    return validator.overrideSymbol(NAME).toUpperCase()
   }
 
-  return {
-    key: util.getRequiredEnv('API_KEY', prefix),
-    symbols: util.getRequiredEnv('SYMBOLS', prefix),
-    provider: provider.toLowerCase(),
+  return async () => {
+    const defaultConfig = config || makeConfig()
+
+    const ws = new IntrinioRealtime({
+      api_key: defaultConfig.apiKey,
+      provider: 'iex',
+    })
+
+    await ws._refreshToken()
+
+    return {
+      connection: {
+        url: ws._makeSocketUrl(),
+      },
+      subscribe: input => ws._makeJoinMessage(getBase(input)),
+      unsubscribe: input => ws._makeLeaveMessage(getBase(input)),
+      subsFromMessage: message => ws._makeJoinMessage(message.payload.ticker),
+      isError: (message: any) => Number(message.TYPE) > 400 && Number(message.TYPE) < 900,
+      filter: message => message.event == 'quote' && message.payload?.type == 'last',
+      toResponse: (wsResponse: any): AdapterResponse => {
+        wsResponse = { ...wsResponse, result: wsResponse?.payload?.price }
+        return Requester.success(undefined, { data: wsResponse })
```

Comment:
- Better give `'1'` as job id instead of `undefined`
- Is it necessary to return the full WS message?

Reply:
Does Requester.success default to `'1'` anyway?

I don't think it's necessary since the it's returning 
```
{
  topic: 'iex:securities:AAPL',
  payload: {
    type: 'last',
    timestamp: 1619704191.36749,
    ticker: 'AAPL',
    size: 50,
    price: 135.01
  },
  event: 'quote'
}
```
and we only need the price. Sounds like it would be good to just return the price to reduce the size of the cache?

Reply:
Right, return the price only then